crates/runtime/src/libcalls.rs

```diff
@@ -402,6 +402,13 @@ pub unsafe extern "C" fn wasmtime_activations_table_insert_with_gc(
     let externref = VMExternRef::clone_from_raw(externref);
     let instance = (*vmctx).instance();
     let (activations_table, module_info_lookup) = (*instance.store()).externref_activations_table();
+
+    // Invariant: all `externref`s on the stack have an entry in the activations
+    // table. So we need to ensure that this `externref` is in the table
+    // *before* we GC, even though `insert_with_gc` will ensure that it is in
+    // the table *after* the GC.
+    activations_table.insert_without_gc(externref.clone());
```

Comment:
This can end up doing a double-insertion between this and `insert_with_gc` below, and it might be good to avoid that? This method is only called when the table is full so `insert_without_gc` is already guaranteed to go straight to insertion in the slow path, so should a specialized method of the table be added for basically this use case?

Reply:
It won't insert into the bump chunk, so I don't think there is anything to worry about here. We just do one more hash lookup than is strictly necessary, but that is fine as this is a slow path anyways.

Reply:
That's true, yeah. Could you update the comment here to indicate that this is doing some extra work but it's the slow path so it should be ok?